packages/api/models/rules.js

```diff
@@ -148,6 +152,23 @@ class Rule extends Manager {
    * @returns {Promise} updated rule item
    */
   async addKinesisEventSource(item) {
+    // use the existing event source mapping if it already exists
+    const listParams = {
+      EventSourceArn: item.rule.value,
+      FunctionName: process.env.kinesisConsumer
+    };
+    const listData = await aws.lambda(listParams).listEventSourceMappings().promise();
+    if (listData.EventSourceMappings && listData.EventSourceMappings.length > 0) {
+      const maps = listData.EventSourceMappings;
```

Comment:
I think we can use `Array.prototype.find` to simplify this loop, something like:
```javascript
if (listData.EventSourceMappings) {
  const eventSourceMappingExists = listData.EventSourceMappings.find((eventSourceMapping) => {
    return eventSourceMapping.EventSourceArn === item.rule.value;
  });
  if (eventSourceMappingExists) {
    item.rule.arn = maps[i].UUID;
    return item;
  }
}
```

I didn't test this so if it doesn't work for some reason this looks fine as-is

Reply:
your find method works!